pootle/apps/pootle_statistics/migrations/0019_convert_accept_sugg_subs.py

```diff
@@ -0,0 +1,68 @@
+# -*- coding: utf-8 -*-
+# Generated by Django 1.10.5 on 2017-03-01 08:04
+from __future__ import unicode_literals
+
+from django.db import migrations
+from django.db.models import F
+
+
+WEB_FLAG = 1
+SUGG_ACCEPT_FLAG = 3
+
+
+def _assoc_state_accept_suggestion_subs(apps):
+    subs = apps.get_model("pootle_statistics.Submission").objects.all()
+    suggestions = apps.get_model("pootle_store.Suggestion").objects
+    accepted = suggestions.filter(state__name="accepted").values(
+        "review_time", "pk", "unit_id", "reviewer")
+    state_acc = subs.filter(type=SUGG_ACCEPT_FLAG).filter(field=3)
+
+    for suggestion in accepted.iterator():
+        suggestion["submitter"] = suggestion["reviewer"]
+        suggestion["creation_time"] = suggestion["review_time"]
+        suggestion_id = suggestion["pk"]
+        del suggestion["review_time"]
+        del suggestion["reviewer"]
+        del suggestion["pk"]
+        state_acc.filter(**suggestion).update(suggestion_id=suggestion_id)
+
+
+def _delete_redundant_accept_sugg_subs(apps):
+    subs = apps.get_model("pootle_statistics.Submission").objects
+    sugg_acc = subs.filter(type=SUGG_ACCEPT_FLAG)
+
+    # these appear to be caused by dupes
+
+    # review time doesnt match
+    sugg_acc.exclude(suggestion__isnull=True).exclude(
+        creation_time=F("suggestion__review_time")).delete()
+
+    # reviewer doesnt match
+    sugg_acc.exclude(suggestion__isnull=True).exclude(
+        submitter=F("suggestion__reviewer")).delete()
+
+    # no suggestion
+    sugg_acc.filter(suggestion__isnull=True).count()
```

Comment:
Why not do this one first? It will save you from having to exclude on the two previous querysets.